fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/service/ReadWriteNonCoreDataServiceImpl.java

```diff
@@ -214,6 +214,185 @@ public List<JsonObject> queryDataTable(String datatable, final String columnFilt
+
+                // Attach the WHERE clause
+                if (baseFilterQuery.length() > 0) {
+                    sqlBuilder.append(" WHERE ").append(baseFilterQuery);
+                    countQueryBuilder.append(" WHERE ").append(baseFilterQuery);
+                }
+
+                // Attach the ORDER
+                attachOrdering(sqlBuilder, pageable.getSort());
+                // Attach the PAGINATION
+                applyPagination(sqlBuilder, pageable);
+
+                SqlRowSet rowSet = jdbcTemplate.queryForRowSet(sqlBuilder.toString());
+
+                // Execute the count Query
+                totalElements.set(jdbcTemplate.queryForObject(countQueryBuilder.toString(), Integer.class));
+                String[] resultColumnNames = resultColumn.split(",");
+
+                while (rowSet.next()) {
+                    extractResults(rowSet, resultColumnNames, results);
+                }
+
+            }
+        }
+        return PageableExecutionUtils.getPage(results, pageable, () -> totalElements.get());
+
+    }
+
+    private void applyPagination(StringBuilder query, Pageable pageable) {
+        if (pageable.isPaged()) {
+            query.append(sqlGenerator.limit(pageable.getPageSize(), (int) pageable.getOffset()));
+        } else {
+            query.append(sqlGenerator.limit(pageable.getPageSize()));
+        }
+
+    }
+
+    private void attachOrdering(StringBuilder query, Sort sort) {
+        if (sort.isSorted()) {
+            query.append(buildOrderByClause(sort.toList()));
+        }
+    }
+
+    private String buildOrderByClause(List<Order> orders) {
+        return orders.stream().map(order -> String.join(" ", order.getProperty(), order.getDirection().name()))
+                .collect(Collectors.joining(", "));
+    }
+
+    private String buildBaseQueryWithFilters(List<ColumnFilter> columnFilters, Map<String, String> dataTableColumnMap,
+            AtomicInteger count) {
+        StringBuilder queryColumnBuilder = new StringBuilder();
+        columnFilters.forEach(column -> {
+
+            String columnName;
+            Object valueFilter;
+            String columnType;
+
+            if (databaseTypeResolver.isPostgreSQL()) {
+                columnName = escapeFieldNames(column.getColumnName());
+                columnType = dataTableColumnMap.get(columnName);
+                valueFilter = callFilteredPgSqlSearch(column.getColumnValue(), columnType);
+            } else {
+                columnName = column.getColumnName();
+                columnType = dataTableColumnMap.get(columnName);
+                valueFilter = callFilteredMysqlSearch(column.getColumnValue(), columnType);
+            }
+
+            if ("EQUALS".equalsIgnoreCase(column.getColumnOperation())) {
+                queryColumnBuilder.append(String.format(" %s %s '%s' ", columnName, "=", valueFilter));
+            } else if ("NOTEQUALS".equalsIgnoreCase(column.getColumnOperation())) {
```

Comment:
it would be more readable if you were using _ for splitting the words: `NOT_EQUALS`